test/unit/error.test.ts

```diff
@@ -89,6 +91,30 @@ describe('MongoErrors', () => {
     });
   });
 
+  describe('MongoSystemError#constructor', () => {
+    context('when the topology description contains a code', () => {
+      it('contains the code as a top level property', () => {
+        const topologyDescription = {
+          error: {
+            code: 123
+          }
+        } as any as EverythingFromDriver.TopologyDescription;
+
+        const error = new MongoSystemError('something went wrong', topologyDescription);
+        expect(error).to.haveOwnProperty('code', 123);
+      });
+    });
+
+    context('when the topology description does not contain a code', () => {
+      it('contains the code as a top level property', () => {
+        const topologyDescription = {} as any as EverythingFromDriver.TopologyDescription;
```

Comment:
I think it might be worthwhile to also test case where the topologyDescription contains an error prop that does not have a code, just to cover all our bases